service/src/main/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperService.java

```diff
@@ -301,45 +326,95 @@ public void addComplyOnBehalfResponsesToDirectionsInStandardDirectionsOrder(Case
                         .map(response -> response.toBuilder().respondingOnBehalfOf("CAFCASS").build())
                         .collect(toList());
 
-                    addResponsesToDirections(responses, caseData.getStandardDirectionOrder().getDirections());
+                    addResponsesToDirections(responses, directionsToComplyWith);
 
                     break;
+
                 case PARENTS_AND_RESPONDENTS:
+                    List<Element<DirectionResponse>> respondentResponses = addValuesToListResponses(
+                        directions, eventId, authorisation, PARENTS_AND_RESPONDENTS);
+
+                    addResponseElementsToDirections(respondentResponses, directionsToComplyWith);
+
+                    break;
                 case OTHERS:
-                    List<Element<DirectionResponse>> elements = addValuesToListResponseDirections(directions);
+                    List<Element<DirectionResponse>> otherResponses = addValuesToListResponses(
+                        directions, eventId, authorisation, OTHERS);
 
-                    addResponseElementsToDirections(elements, caseData.getStandardDirectionOrder().getDirections());
+                    addResponseElementsToDirections(otherResponses, directionsToComplyWith);
 
                     break;
             }
         });
     }
 
-    private List<Element<DirectionResponse>> addValuesToListResponseDirections(List<Element<Direction>> directions) {
-        AtomicReference<UUID> id = new AtomicReference<>();
+    //TODO: refactor of addCourtAssigneeAndDirectionId would remove dependency on eventId.
+    private List<Element<DirectionResponse>> addValuesToListResponses(List<Element<Direction>> directions,
+                                                                      ComplyOnBehalfEvent eventId,
+                                                                      String authorisation,
+                                                                      DirectionAssignee assignee) {
+        final UUID[] id = new UUID[1];
```

Comment:
Do we prefer this or `AtomicReference<UUID>`

Reply:
AtomicReference does a lot. UUID array is much simpler. @akomisarek asked me to do this before, I finally caved 😂